data-plane/tests/src/test/java/dev/knative/eventing/kafka/broker/tests/DataPlaneTest.java

```diff
@@ -260,15 +260,16 @@ public void execute(final Vertx vertx, final VertxTestContext context) throws Ex
   @AfterAll
   public static void teardown(final Vertx vertx) {
 
-    // TODO figure out why shutdown times out Vertx context even with timeout increased.
-    // kafkaCluster.shutdown();
+    if (kafkaCluster != null) {
+      kafkaCluster.shutdown();
+      kafkaCluster = null;
+      if (!dataDir.delete()) {
+        dataDir.deleteOnExit();
+      }
+    }
 
     vertx.undeploy(consumerDeployerVerticle.deploymentID());
```

Comment:
This way consumers stil running will fail, right?

I think we should rather have the `vertx.undeploy` block here (using `toCompletionStage().toCompletableFuture().get()`) and then shutdown the kafka cluster